File: src/main/java/dev/mantasboro/trakt5/entities/SyncShow.java
package dev.mantasboro.trakt5.entities;

import dev.mantasboro.trakt5.enums.Rating;
import org.threeten.bp.OffsetDateTime;

import java.util.ArrayList;
import java.util.List;

public class SyncShow {

    public ShowIds ids;
    public List<SyncSeason> seasons;

    public OffsetDateTime collected_at;
    public OffsetDateTime watched_at;
    public OffsetDateTime rated_at;
    public Rating rating;

    public SyncShow id(ShowIds id) {
        this.ids = id;
        return this;
    }

    public SyncShow seasons(List<SyncSeason> seasons) {
        this.seasons = seasons;
        return this;
    }

    public SyncShow seasons(SyncSeason season) {
        ArrayList<SyncSeason> list = new ArrayList<>(1);
        list.add(season);
        return seasons(list);
    }

    public SyncShow collectedAt(OffsetDateTime collectedAt) {
        this.collected_at = collectedAt;
        return this;
    }

    public SyncShow watchedAt(OffsetDateTime watchedAt) {
        this.watched_at = watchedAt;
        return this;
    }

    public SyncShow ratedAt(OffsetDateTime ratedAt) {
        this.rated_at = ratedAt;
        return this;
    }

    public SyncShow rating(Rating rating) {
        this.rating = rating;
        return this;
    }

}

File: src/main/java/dev/mantasboro/trakt5/entities/SyncShow.kt
package dev.mantasboro.trakt5.entities

import dev.mantasboro.trakt5.enums.Rating
import org.threeten.bp.OffsetDateTime

class SyncShow {
    var ids: ShowIds? = null
    var seasons: List<SyncSeason>? = null
    var collected_at: OffsetDateTime? = null
    var watched_at: OffsetDateTime? = null
    var rated_at: OffsetDateTime? = null
    var rating: Rating? = null
    fun id(id: ShowIds?): SyncShow {
        ids = id
        return this
    }

    fun seasons(seasons: List<SyncSeason>?): SyncShow {
        this.seasons = seasons
        return this
    }

    fun seasons(season: SyncSeason): SyncShow {
        val list = ArrayList<SyncSeason>(1)
        list.add(season)
        return seasons(list)
    }

    fun collectedAt(collectedAt: OffsetDateTime?): SyncShow {
        collected_at = collectedAt
        return this
    }

    fun watchedAt(watchedAt: OffsetDateTime?): SyncShow {
        watched_at = watchedAt
        return this
    }

    fun ratedAt(ratedAt: OffsetDateTime?): SyncShow {
        rated_at = ratedAt
        return this
    }

    fun rating(rating: Rating?): SyncShow {
        this.rating = rating
        return this
    }
}
